openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -792,6 +792,14 @@ def _resolve_parameter_handler_order(self):
     # TODO: How do we know if the system is periodic or not?
     # TODO: Should we also accept a Molecule as an alternative to a Topology?
 
+
+    def configure_vacuum(self):
```

Comment:
I'm uncertain we want to expose an API point like this. Following my conversation with @j-wags, the `ForceField` should have a recipe for how to process `Topology` into a `System`, and the `Topology` should know whether it is supposed to represent molecule or cluster in vacuum or in a condensed phase system. We can certainly add API methods to the `Topology` object that let us manipulate what system it is supposed to represent, but that should _not_ happen at the `ForceField` level, since the force field is just the recipe on how to apply parameters to a given description of a system.

Reply:
`configure_vacuum` is my idea for fixing a major roadblock in terms of user experience. It's likely that I've been looking at this specific issue so long that I'm overlooking a better solution. But here's my reasoning:

`configure_vacuum` has several flaws, including setting things to opaque "vacuum default" values, and changing the API of every other ParameterHandler.

However, right now, our `smirnoff99Frosst` file specifies parameters specific to condensed-phase systems (for example, `PME` electrostatics). We will, presumably, put out examples for how to run vacuum phase simulations. If we don't include `configure_vacuum` I'm concerned that these vacuum simulation examples will look like:

```
ff = ForceField('smirnoff99Frosst.offxml')
ES_handler = ff.get_handler('Electrostatics')
ES_handler.method = 'Coulomb'
del ES_handler._cutoff
del ES_handler._switch
vdw_handler = ff.get_handler('vdW')
vdw_handler.long_range_dispersion = 'None'
del vdw_handler._cutoff

topology = Topology.from_openmm(pdbfile.topology, unique_molecules=molecules, load_box_vectors=False)
```

We could make the API more convenient to a degree, but the fact remains that the user would have to manually change a number of settings to get our default OFFXML files to run a vacuum simulation.

I had considered two other ideas as well:
* to automatically have the `ForceField`/`ParameterHandler`s switch to vacuum-compatible settings if a vacuum topology were passed in, but, this is as bad as `configure_vacuum`, while being even more opaque.
* to have separate `smirnoff99Frosst_condensed` and `_vacuum` OFFXMLs, but I'm hesitant to double the number of files we keep around. 

I think it's likely that there's a better way to handle this, but I wasn't able to think of it. Advice would be appreciated!

Reply:
The way we *should* handle this is for the forcefield OFF XML definition file to include information on how both vacuum and condensed phase systems should be treated. However, in our conversation, it became apparent there was currently only one way that vacuum systems should be treated, and that if there were more variety we might need to expose in future, we could do this by extending the spec.

My take away from our discussion was that the Topology object would know whether it represented a vacuum or condensed phase system, and that the OFF XML spec contained information on how the condensed phase representation should work (with vacuum being implicit but extensible in future). We would then exclusively have the user be concerned with what the Topology object represented for the system in condensed or vacuum phases, and the ForceField would just apply the protocol for either vacuum or condensed phases.

Reply:
But what about something like transforming `method` and `long_range_dispersion` into properties that automatically configure `cutoff` and `switch_width` accordingly?

That would reduce it to
```python
ff = ForceField('smirnoff99Frosst.offxml')
ff.get_handler('Electrostatics').method = 'Coulomb'
ff.get_handler('vdW').long_range_dispersion = None
```
Even better, I think, would be to not delete entirely the attributes and have, but have a property adopt the correct behavior. For example for `cutoff` we can have
```python
@property
def cutoff(self):
    if self.method == 'Coulomb':
        return None  # or alternatively raise AttributeError
    return self._cutoff
```
the effect would be the same, but if you switch the `method` back to `PME`, then the original cutoff loaded from the force field XML will still be in memory.

Reply:
@jchodera 
> My take away from our discussion was that the Topology object would know whether it represented a vacuum or condensed phase system, and that the OFF XML spec contained information on how the condensed phase representation should work (with vacuum being implicit but extensible in future).

I see. This is a good design decision, and I hadn't fully understood it before. Three action items come from this:

1) I'll implement a switchover to vacuum settings if the Topology is provided without a periodic box.
2) I'll remove `configure_vacuum` from the public API
3) I'll open up an issue to discuss how we can encode nonperiodic and periodic defaults in the same OFFXML file (right now it happens to be easy because we only have one setting for vacuum, but it could get more complicated later)

Reply:
@andrrizzi 

> What about something like transforming method and long_range_dispersion into properties that automatically configure cutoff and switch_width accordingly?

I think that's a good idea once I have a better understanding of all of the desired logic here. Given the timeline, I'm aiming to do "space shuttle" coding, where each situation is handled more-or-less explicitly. This way we can visually ensure the logic is right, and make the API friendlier later.

That said, I'm rethinking the current data structure implementation now that I understand that a single OFFXML should contain both periodic and nonperiodic settings.

Reply:
> I'll open up an issue to discuss how we can encode nonperiodic and periodic defaults

Let's be clear about calling them "vacuum" and "condensed" since, as we discussed, we can use GBSA to simulate the condensed phase but the system would not be periodic.

> What about something like transforming method and long_range_dispersion into properties that automatically configure cutoff and switch_width accordingly?

@andrrizzi's suggestion is a great one if we want people to be able to change the nonbonded treatment via the API and have validators not cause an error because you would have to set cutoff and nonbonded treatment simultaneously. I think the idea of defaults is probably the only sensible solution.